packages/voting/graphql/resolvers/_mutations/submitAnswer.js

```diff
@@ -0,0 +1,131 @@
+const { ensureSignedIn } = require('@orbiting/backend-modules-auth')
+const {
+  findById,
+  ensureReadyToSubmit,
+  transformQuestion
+} = require('../../../lib/Questionnaire')
+const { graphql: { resolvers: { queries: { document: getDocument } } } } = require('@orbiting/backend-modules-documents')
+
+module.exports = async (_, { answer: { questionId, payload } }, context) => {
+  const { pgdb, user: me, t, req } = context
+  ensureSignedIn(req, t)
+
+  const transaction = await pgdb.transactionBegin()
+  try {
+    const now = new Date()
+    const question = await transaction.public.questions.findOne({ id: questionId })
+    if (!question) {
+      throw new Error(t('api/questionnaire/question/404'))
+    }
+
+    const questionnaire = await findById(question.questionnaireId, transaction)
+    await ensureReadyToSubmit(questionnaire, me.id, now, transaction, t)
+
+    let emptyAnswer = false
+    if (!payload) {
+      emptyAnswer = true
+    } else {
+      // validate payload
+      if (payload.value === undefined || payload.value === null) {
+        throw new Error(t('api/questionnaire/answer/empty'))
+      }
+
+      const { value } = payload
+      if (question.type === 'Text') {
```

Comment:
I find `switch` easier to read than `if else` chains. The validation logic could be packed into functions that return a boolean to be assigned to `emptyAnswer`.

```
switch(question.type) {
  case 'Text':
    emptyAnswer = validateText(value)
    break
  ...
  default:
    throw new Error(t('api/questionnaire/question/type/404', { type: question.type }))
}
```

Reply:
Thanks for the feedback. I grouped question validate/result into heir own files here: https://github.com/orbiting/backends/pull/130/commits/10c980dcded82fc75fc127719bda4df8ed530bc5